src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -183,17 +197,52 @@ public static bool UpdateInstallerNodes(InstallerManifest installerManifest, IEn
             // Update previous installers with parsed data from downloaded packages
             foreach (var newInstaller in newInstallers)
             {
-                // Find a match for the current installer in the list of installers remaining to be updated.
-                var matchingExistingInstaller = existingInstallers.SingleOrDefault(i =>
-                    (i.InstallerType ?? installerManifest.InstallerType) == newInstaller.InstallerType &&
+                var archGuess = GetArchFromUrl(newInstaller.InstallerUrl);
+
+                if (archGuess != null && archGuess != newInstaller.Architecture)
+                {
+                    ArchMismatches.Add((newInstaller.InstallerUrl, (InstallerArchitecture)archGuess, newInstaller.Architecture));
+                    newInstaller.Architecture = (InstallerArchitecture)archGuess;
+                }
```

Comment:
Doesn't this mess up the logging for unmatched/multipleMatchedInstallers? I believe you need to return a tuple for those too...

Reply:
Yes you're right, I created a detectedArch object to keep track of the architectures detected from the url and binary. I believe the logging should be correct now that the the different detected architectures are stored separately